data/serializer/src/main/java/tech/pegasys/teku/api/schema/BeaconBlockBody.java

```diff
@@ -91,36 +91,38 @@ public BeaconBlockBody(
-              .graffiti(graffiti)
-              .attestations(
-                  attestations.stream()
-                      .map(attestation -> attestation.asInternalAttestation(spec))
-                      .collect(schema.getAttestationsSchema().collector()))
-              .proposerSlashings(
-                  proposer_slashings.stream()
-                      .map(ProposerSlashing::asInternalProposerSlashing)
-                      .collect(schema.getProposerSlashingsSchema().collector()))
-              .attesterSlashings(
-                  attester_slashings.stream()
-                      .map(slashing -> slashing.asInternalAttesterSlashing(spec))
-                      .collect(schema.getAttesterSlashingsSchema().collector()))
-              .deposits(
-                  deposits.stream()
-                      .map(Deposit::asInternalDeposit)
-                      .collect(schema.getDepositsSchema().collector()))
-              .voluntaryExits(
-                  voluntary_exits.stream()
-                      .map(SignedVoluntaryExit::asInternalSignedVoluntaryExit)
-                      .collect(schema.getVoluntaryExitsSchema().collector()));
-        });
+    return schema
+        .createBlockBody(
+            builder -> {
+              builderRef.accept(builder);
+              builder
+                  .randaoReveal(randao_reveal.asInternalBLSSignature())
+                  .eth1Data(
+                      new tech.pegasys.teku.spec.datastructures.blocks.Eth1Data(
+                          eth1_data.deposit_root, eth1_data.deposit_count, eth1_data.block_hash))
+                  .graffiti(graffiti)
+                  .attestations(
+                      attestations.stream()
+                          .map(attestation -> attestation.asInternalAttestation(spec))
+                          .collect(schema.getAttestationsSchema().collector()))
+                  .proposerSlashings(
+                      proposer_slashings.stream()
+                          .map(ProposerSlashing::asInternalProposerSlashing)
+                          .collect(schema.getProposerSlashingsSchema().collector()))
+                  .attesterSlashings(
+                      attester_slashings.stream()
+                          .map(slashing -> slashing.asInternalAttesterSlashing(spec))
+                          .collect(schema.getAttesterSlashingsSchema().collector()))
+                  .deposits(
+                      deposits.stream()
+                          .map(Deposit::asInternalDeposit)
+                          .collect(schema.getDepositsSchema().collector()))
+                  .voluntaryExits(
+                      voluntary_exits.stream()
+                          .map(SignedVoluntaryExit::asInternalSignedVoluntaryExit)
+                          .collect(schema.getVoluntaryExitsSchema().collector()));
+            })
+        .join();
```

Comment:
It seems to me that here it might be only place where makes sense to have the `join()` since we are providing an immediate completed future for header and payload. But I need to double-check

Reply:
Given we want to get rid of these API classes I'd say this is probably a good point to use a join yeah. And yes we know it will already be completed anyway which makes me feel more ok with it too.